build-logic/src/main/kotlin/terasology-module.gradle.kts

```diff
@@ -145,7 +148,7 @@ tasks.named("processResources") {
 }
 
 tasks.named("compileJava") {
-    dependsOn("processResources")
+    dependsOn("processResources", ":engine:compileJmhJava")
```

Comment:
the error message was:
```
> Task :engine:jar
> Task :modules:CoreAssets:compileJava NO-SOURCE
> Task :modules:BiomesAPI:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':modules:BiomesAPI:compileJava' (type 'JavaCompile').
  - Gradle detected a problem with the following location: '/srv/http/repos/terasology/engine/build/classes'.
    
    Reason: Task ':modules:BiomesAPI:compileJava' uses this output of task ':engine:compileJmhJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':engine:compileJmhJava' as an input of ':modules:BiomesAPI:compileJava'.
      2. Declare an explicit dependency on ':engine:compileJmhJava' from ':modules:BiomesAPI:compileJava' using Task#dependsOn.
      3. Declare an explicit dependency on ':engine:compileJmhJava' from ':modules:BiomesAPI:compileJava' using Task#mustRunAfter.
    
    For more information, please refer to https://docs.gradle.org/8.5/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.

```

Reply:
thanks for the feedback on discord. to make the engine jar not contain the jmh files i added
```
    exclude("src/jmh")
```
which then allowed to compile it without all these undesired dependency spread out.